### ISSUE TITLE
Not closed tag inside if blocks are not highlighted by parser

### ISSUE THREAD

**Issue by marcalexiei** [ractivejs/ractive#3051]
## Description:
If you include a component inside an _if_ block without closing tag the parser compile template wihtout error and when the condition is _true_ the if content is rendered correctly in the page.
However when condition is _false_  the content inside _else_ block is not rendered in the page.
Content after both blocks is rendered correctly in both case.

Another issue appears with not closed HTML tags: if condition is _true_ both _if_ and _else_ blocks are displayed.

Should parser throw an error on those situations?
 
## Versions affected:
- 0.9.x
- edge

## Platforms affected:
- All

## Reproduction:
*Component scenario*
1. Open https://jsfiddle.net/hjgmp41r/
2. When _showMe_ is _false_ "I'm inside the else" is not displayed.

```html
<div id="main"></div>

<script type="text" id="template">
    <h1>Hello {{who}}!</h1>
    <p>We are ready for some bugs!</p>

    showMe: {{showMe}}

    <button on-click="@.toggle('showMe')">Toggle</button>

    <div class="comp">
      {{#if showMe}}
        <MyComp>
      {{else}}
        I'm inside the else
      {{/if}}
    </div>

    <p>other content</p>
</script>
```

```javascript
Ractive.components.MyComp = Ractive.extend({
    template: 'Hi, I\'m a template!!!!!',
});

const r = window.r = new Ractive({
    el: '#main',
    template: '#template',
    data: {
        showMe: false,
    }
});
```

*HTML scenario*
1. Open https://jsfiddle.net/axxw71xp/
2. When _showMe_ is _true_ both blocks are displayed.

```html
<div id="main"></div>

<script type="text" id="template">
 <h1>Hello {{who}}!</h1>
    <p>We are ready for some bugs!</p>

    showMe: {{showMe}}

    <button on-click="@.toggle('showMe')">Toggle</button>

    <div class="comp">
      {{#if showMe}}
       <div>showMe is true
      {{else}}
        I'm inside the else
      {{/if}}
    </div>

    <p>other content</p>
</script>
```

```javascript
Ractive.components.MyComp = Ractive.extend({
	template: 'Hi, I\'m a template!!!!!',
});

const r = window.r = new Ractive({
  el: '#main',
  template: '#template',
  data: {
  	showMe: false,
  }
});
```

**Comment by evs-chris** [comment]
TL;DR the `{{else}}` is being eaten as a plain interpolator because the component context owns the parser when it appears.

The issue is kinda due to the tacked-on alternate branching of handlebars. If `else` was actually a block directive, it would be completely obvious where to end the component, but as it's just a special-cased interpolator that the `if` block handles, the component is free to eat it. That leaves the component `content` partials as `{{else}}I'm inside the else` where the interpolator will most likely be empty unless you do `this.set('else', ...)` in the component.

That _could_ be worked around in the parser, but I don't think it's exactly a trivial change. I'll flag it as an enhancement, and it would be a good first PR for anyone wanting an introduction to the parser :wink:

**Comment by ghost** [comment]
Strange... I would think Ractive would throw an error due to one of the following:

- Component tag is open but the next thing that closed is an `{{/if}}`. 
- An `{{else}}` was encountered without a preceding `{{#if}}` in the component body.

This is probably a bug as well as enhancement to error messaging.

**Comment by evs-chris** [comment]
Ractive doesn't know what's a component and what's a regular element at parse time, though, and html is very (perhaps a bit too) forgiving about closing tags. The closing block scope of the `if` also implicitly closes the tag, that just happens to be a component at runtime in this case.

That `{{else}}` has no special treatment outside of section processing, and then only if it's an immediate child of the section, can be remedied. It needs to be added as a block scope terminator for other block scoped items as well, which I think is not a particularly simple change.
